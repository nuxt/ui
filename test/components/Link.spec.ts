import { describe, it, expect } from 'vitest'
import { ULink as Link } from '#components'
import type { LinkProps, LinkSlots } from '../../src/runtime/components/Link.vue'
import ComponentRender from '../component-render'

describe('Link', () => {
  it.each([
    // Props
    ['with as', { props: { as: 'div' } }],
    ['with to', { props: { to: '/' } }],
    ['with type', { props: { type: 'submit' as const } }],
    ['with disabled', { props: { disabled: true } }],
    ['with raw', { props: { raw: true } }],
    ['with class', { props: { class: 'font-medium' } }],
    ['with activeClass', { props: { active: true, activeClass: 'text-[var(--ui-text-highlighted)]' } }],
    ['with inactiveClass', { props: { active: false, inactiveClass: 'hover:text-[var(--ui-primary)]' } }],
    // Slots
    ['with default slot', { slots: { default: () => 'Default slot' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: LinkProps, slots?: Partial<LinkSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, Link)
    expect(html).toMatchSnapshot()
  })
})
