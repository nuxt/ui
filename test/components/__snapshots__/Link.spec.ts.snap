// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > renders with activeClass correctly 1`] = `"<button type="button" class="focus-visible:outline-[var(--ui-primary)] text-[var(--ui-text-highlighted)]"></button>"`;

exports[`Link > renders with as correctly 1`] = `"<div class="focus-visible:outline-[var(--ui-primary)] text-[var(--ui-text-muted)] hover:text-[var(--ui-text)] transition-colors"></div>"`;

exports[`Link > renders with class correctly 1`] = `"<button type="button" class="focus-visible:outline-[var(--ui-primary)] text-[var(--ui-text-muted)] hover:text-[var(--ui-text)] transition-colors font-medium"></button>"`;

exports[`Link > renders with default slot correctly 1`] = `"<button type="button" class="focus-visible:outline-[var(--ui-primary)] text-[var(--ui-text-muted)] hover:text-[var(--ui-text)] transition-colors">Default slot</button>"`;

exports[`Link > renders with disabled correctly 1`] = `"<button type="button" disabled="" class="focus-visible:outline-[var(--ui-primary)] text-[var(--ui-text-muted)] hover:text-[var(--ui-text)] transition-colors cursor-not-allowed opacity-75"></button>"`;

exports[`Link > renders with inactiveClass correctly 1`] = `"<button type="button" class="focus-visible:outline-[var(--ui-primary)] text-[var(--ui-text-muted)] transition-colors hover:text-[var(--ui-primary)]"></button>"`;

exports[`Link > renders with raw correctly 1`] = `"<button type="button" class=""></button>"`;

exports[`Link > renders with to correctly 1`] = `"<a href="/" class="focus-visible:outline-[var(--ui-primary)] text-[var(--ui-text-muted)] hover:text-[var(--ui-text)] transition-colors"></a>"`;

exports[`Link > renders with type correctly 1`] = `"<button type="submit" class="focus-visible:outline-[var(--ui-primary)] text-[var(--ui-text-muted)] hover:text-[var(--ui-text)] transition-colors"></button>"`;
